specification/elastic/resource-manager/Microsoft.Elastic/preview/2022-09-01-preview/elastic.json

```diff
@@ -0,0 +1,2308 @@
+            "$ref": "#/definitions/ElasticTrafficFilter"
+          }
+        }
+      }
+    },
+    "ElasticTrafficFilter": {
+      "description": "Elastic traffic filter object",
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Id of the elastic filter"
+        },
+        "name": {
+          "type": "string",
+          "description": "Name of the elastic filter"
+        },
+        "description": {
+          "type": "string",
+          "description": "Description of the elastic filter"
+        },
+        "region": {
+          "type": "string",
+          "description": "Region of the elastic filter"
+        },
+        "type": {
+          "enum": [
+            "ip",
+            "azure_private_endpoint"
+          ],
+          "type": "string",
+          "description": "Type of the elastic filter",
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "type"
+          }
+        },
+        "includeByDefault": {
+          "type": "boolean",
+          "description": "IncludeByDefault for the elastic filter"
+        },
+        "rules": {
+          "type": "array",
+          "description": "Rules in the elastic filter",
+          "items": {
+            "$ref": "#/definitions/ElasticTrafficFilterRule"
+          }
+        }
+      }
+    },
+    "ElasticTrafficFilterRule": {
+      "description": "Elastic traffic filter rule object",
+      "type": "object",
+      "properties": {
+        "source": {
+          "type": "string",
+          "description": "IP of the elastic filter rule"
+        },
+        "description": {
+          "type": "string",
+          "description": "Description of the elastic filter rule"
+        },
+        "azureEndpointGuid": {
+          "type": "string",
+          "description": "Guid of Private Endpoint in the elastic filter rule"
+        },
+        "azureEndpointName": {
+          "type": "string",
+          "description": "Name of the Private Endpoint in the elastic filter rule"
+        },
+        "id": {
+          "type": "string",
+          "description": "Id of the elastic filter rule"
+        }
+      }
+    },
+    "ElasticMonitorUpgrade": {
+      "description": "Upgrade elastic monitor version",
+      "type": "object",
+      "properties": {
+        "version": {
+          "type": "string",
+          "description": "Version to which the elastic monitor should be upgraded to"
+        }
+      }
+    }
+  },
+  "parameters": {
+    "ApiVersionParameter": {
+      "name": "api-version",
+      "in": "query",
+      "description": "The API version to be used with the HTTP request.",
+      "required": true,
+      "type": "string",
+      "enum": [
+        "2020-07-01-preview",
+        "2020-07-01",
+        "2021-09-01-preview",
+        "2021-10-01-preview",
+        "2022-05-05-preview",
+        "2022-07-01-preview",
+        "2022-09-01-preview"
```

Comment:
This is not a proper way of represeting the API version parameter. Each swagger is already linked ot a single API version. Just use the common type definition for API version instead

Reply:
@rkmanda this is how we've been always adding a new version. Could you please point us to the documentation or any reference which describes how this is supposed to be added in the right way?

Reply:
See common type: [https://github.com/Azure/azure-rest-api-specs-pr/blob/d89e212da2cc34206fe711f44dfcb6f8fdece2a1/specification/common-types/resource-management/v3/types.json#L630](https://github.com/Azure/azure-rest-api-specs-pr/blob/d89e212da2cc34206fe711f44dfcb6f8fdece2a1/specification/common-types/resource-management/v3/types.json#L630)

Reply:
Addressed